server/pulp/server/async/tasks.py

```diff
@@ -450,13 +450,15 @@ def __call__(self, *args, **kwargs):
         This overrides PulpTask's __call__() method. We use this method
         for task state tracking of Pulp tasks.
         """
-        # Check task status and skip running the task if task state is 'canceled'.
+        # Check task status and skip running the task if task state is in one of complete states.
         try:
             task_status = TaskStatus.objects.get(task_id=self.request.id)
         except DoesNotExist:
             task_status = None
-        if task_status and task_status['state'] == constants.CALL_CANCELED_STATE:
-            _logger.debug("Task cancel received for task-id : [%s]" % self.request.id)
+        if task_status and task_status['state'] in constants.CALL_COMPLETE_STATES:
+            _logger.debug(
+                "Task is in the %s state, task-id : [%s]" % (task_status['state'], self.request.id)
```

Comment:
@bmbouter , any suggestions for the debug message are welcome.

Reply:
This looks good to me.